src/wp-includes/functions.php

```diff
@@ -190,6 +192,19 @@ function wp_maybe_decline_date( $date ) {
 
 			$date = preg_replace( $months, $months_genitive, $date );
 		}
+
+		// Match a format like 'F jS' or 'F j' and change it to 'j F'
+		if ( preg_match( '#^[^\d ]+ \d{1,2}(st|nd|rd|th)? #u', trim( $date ) ) ) {
+			foreach ( $months as $key => $month ) {
+				$months[ $key ] = '#' . $month . ' (\d{1,2})(st|nd|rd|th)?#u';
+			}
+
+			foreach ( $months_genitive as $key => $month ) {
+				$months_genitive[ $key ] = '$1 ' . $month;
+			}
+
+			$date = preg_replace( $months, $months_genitive, $date );
+		}
```

Comment:
This whole section is not part of 45611.
See https://core.trac.wordpress.org/changeset/45611/trunk/src/wp-includes/functions.php